src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,88 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final double DEFAULT_RADIUS = 10.0;
+    private static final String DEFAULT_COLOR = Color.WHITE.name();
+    private static final int MAX_SIDE = 100;
+    private static final int FIGURE_TYPE_COUNT = 5;
+
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureType = random.nextInt(FIGURE_TYPE_COUNT);
+        switch (figureType) {
+            case 0:
+                return getRandomSquare();
+            case 1:
+                return getRandomRectangle();
+            case 2:
+                return getRandomCircle();
+            case 3:
+                return getRandomRightTriangle();
+            case 4:
+                return getRandomIsoscelesTrapezoid();
+            default:
+                throw new IllegalStateException("Unexpected value: " + figureType);
+        }
+    }
+
+    public static Figure getDefaultFigure() {
+        return new Circle(DEFAULT_RADIUS, DEFAULT_COLOR);
+    }
+
+    public double getRandomSideLength() {
+        return random.nextDouble(MAX_SIDE);
+    }
+
+    public Square getSquare() {
```

Comment:
```suggestion
    private Square getSquare() {
```

same for other methods for creating figures